packages/stream-node/writer.js

```diff
@@ -49,10 +55,16 @@ export const makeNodeWriter = writer => {
   const nodeWriter = harden({
     /** @param {Uint8Array} value */
     async next(value) {
+      !finalized || Fail`Cannot write into closed Node stream`;
+
       return Promise.race([
         finalIteration,
-        new Promise(resolve => {
-          if (!writer.write(value)) {
+        new Promise((resolve, reject) => {
+          if (
+            !writer.write(value, err => {
+              if (err) reject(err);
+            })
```

Comment:
Is there no other way to observe the error? Node.js I think rightly avoids allocating a closure in the general case.

I’m fine with this change, but it’d be nice to find another way in the future. We’re not close to benching this to vet a decision.

Reply:
(and we’re not close to zero-alloc since we are living at the heady heights of a promise abstraction)

Reply:
Coming to think of it, if we’re going to create a closure either way, the callback should indicate drain in the non-error case so maybe this `if` block is superfluous now.

Reply:
I thought of this but I believe we want to keep accepting writes until the target writable stream is full, right? The callback is invoked on error, or once the data has been flushed. If we were to use the callback instead of checking the return value of `write`, we would basically make the underlying queue inexistent. This is a case where the AsyncIterator abstraction needs to make a choice. Since we do not introduce a queue ourselves, passing through the underlying queue is still correct.

Reply:
Ah, I misinterpreted the docs! The callback isn’t a drainback as I’d assumed.